crates/wasi-nn/src/backend/mod.rs

```diff
@@ -39,7 +47,20 @@ pub enum BackendError {
     #[error("Failed while accessing guest module")]
     GuestAccess(#[from] GuestError),
     #[error("The backend expects {0} buffers, passed {1}")]
-    InvalidNumberOfBuilders(u32, u32),
+    InvalidNumberOfBuilders(usize, usize),
     #[error("Not enough memory to copy tensor data of size: {0}")]
     NotEnoughMemory(usize),
 }
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub(crate) enum BackendKind {
+    OpenVINO,
+}
+impl From<u8> for BackendKind {
+    fn from(value: u8) -> Self {
+        match value {
+            0 => BackendKind::OpenVINO,
+            _ => panic!("invalid backend"),
```

Comment:
Does this impl need to exist? Where does it get called from?

Reply:
Yeah, it's used by `witx.rs` but I guess the more-correct thing is a `impl From<gen::types::GraphEncoding> for BackendKind`. I'll switch to that.